SDL2 build failure

**Describe the bug**
On Fedora 32 running ./build-sdl2 fails since the SDLNet change.
```
/usr/bin/ld: hardware/serialport/libserial.a(misc_util.o): in function `TCPClientSocket::TCPClientSocket(int)':
/home/robert/github/dosbox-x/src/hardware/serialport/misc_util.cpp:57: undefined reference to `SDLNet_GetError'
/usr/bin/ld: hardware/serialport/libserial.a(misc_util.o): in function `TCPClientSocket::TCPClientSocket(_TCPsocket*)':
/home/robert/github/dosbox-x/src/hardware/serialport/misc_util.cpp:114: undefined reference to `SDLNet_GetError'
/usr/bin/ld: hardware/serialport/libserial.a(misc_util.o): in function `TCPClientSocket::TCPClientSocket(char const*, unsigned short)':
/home/robert/github/dosbox-x/src/hardware/serialport/misc_util.cpp:139: undefined reference to `SDLNet_GetError'
/usr/bin/ld: hardware/serialport/libserial.a(misc_util.o): in function `TCPServerSocket::TCPServerSocket(unsigned short)':
/home/robert/github/dosbox-x/src/hardware/serialport/misc_util.cpp:298: undefined reference to `SDLNet_GetError'
/usr/bin/ld: gui/libgui.a(midi.o): in function `MidiHandler_timidity::timidity_write_data(void const*, int)':
/home/robert/github/dosbox-x/src/gui/midi_timidity.h:326: undefined reference to `SDLNet_GetError'
/usr/bin/ld: gui/libgui.a(midi.o):/home/robert/github/dosbox-x/src/gui/midi_timidity.h:326: more undefined references to `SDLNet_GetError' follow
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:470: dosbox-x] Error 1
make[3]: Leaving directory '/home/robert/github/dosbox-x/src'
make[2]: *** [Makefile:530: all-recursive] Error 1
make[2]: Leaving directory '/home/robert/github/dosbox-x/src'
make[1]: *** [Makefile:390: all-recursive] Error 1
make[1]: Leaving directory '/home/robert/github/dosbox-x'
make: *** [Makefile:327: all] Error 2
```

@joncampbell123 

I just pulled the latest changes, and SDL2 build still fails as it did before.
```
/usr/bin/ld: hardware/serialport/libserial.a(misc_util.o): in function `TCPClientSocket::TCPClientSocket(int)':
/home/robert/github/dosbox-x/src/hardware/serialport/misc_util.cpp:57: undefined reference to `SDLNet_GetError'
/usr/bin/ld: hardware/serialport/libserial.a(misc_util.o): in function `TCPClientSocket::TCPClientSocket(_TCPsocket*)':
/home/robert/github/dosbox-x/src/hardware/serialport/misc_util.cpp:114: undefined reference to `SDLNet_GetError'
/usr/bin/ld: hardware/serialport/libserial.a(misc_util.o): in function `TCPClientSocket::TCPClientSocket(char const*, unsigned short)':
/home/robert/github/dosbox-x/src/hardware/serialport/misc_util.cpp:139: undefined reference to `SDLNet_GetError'
/usr/bin/ld: hardware/serialport/libserial.a(misc_util.o): in function `TCPServerSocket::TCPServerSocket(unsigned short)':
/home/robert/github/dosbox-x/src/hardware/serialport/misc_util.cpp:298: undefined reference to `SDLNet_GetError'
/usr/bin/ld: gui/libgui.a(midi.o): in function `MidiHandler_timidity::timidity_write_data(void const*, int)':
/home/robert/github/dosbox-x/src/gui/midi_timidity.h:326: undefined reference to `SDLNet_GetError'
/usr/bin/ld: gui/libgui.a(midi.o):/home/robert/github/dosbox-x/src/gui/midi_timidity.h:326: more undefined references to `SDLNet_GetError' follow
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:470: dosbox-x] Error 1
make[3]: Leaving directory '/home/robert/github/dosbox-x/src'
make[2]: *** [Makefile:530: all-recursive] Error 1
make[2]: Leaving directory '/home/robert/github/dosbox-x/src'
make[1]: *** [Makefile:390: all-recursive] Error 1
make[1]: Leaving directory '/home/robert/github/dosbox-x'
make: *** [Makefile:327: all] Error 2
```

And just to be sure it was nothing with my local copy, I deleted it, and re-cloned it. But it fails the same way.

@joncampbell123 

I just pulled the latest again and this is the result. I guess I could have created a new issue but this seems to be related to the issue experienced by, @rderooy 

Thanks for your wonderful work! :)

g++  -g -std=gnu++11  -O2  -Wall   -Wextra   -Wunused   -pedantic   -Wlogical-op   -Wsign-promo   -Wconversion-null   -Wno-deprecated-declarations   -Wno-implicit-fallthrough   -Wno-strict-aliasing   -Wno-missing-field-initializers   -Wno-format-zero-length   -Wno-address-of-packed-member   -Wno-int-to-void-pointer-cast  -D_XOPEN_SOURCE=700 -D_POSIX_C_SOURCE=200809L -I/home/pi/code/dosbox-x/vs2015/sdlnet/linux-host/include -I/home/pi/code/dosbox-x/vs2015/sdlnet/linux-host/include/SDL   -L/home/pi/code/dosbox-x/vs2015/sdlnet/linux-host/lib -o dosbox-x dosbox.o  debug/libdebug.a dos/libdos.a shell/libshell.a builtin/libbuiltin.a ints/libints.a misc/libmisc.a hardware/serialport/libserial.a hardware/parport/libparallel.a libs/porttalk/libporttalk.a gui/libgui.a libs/gui_tk/libgui_tk.a hardware/libhardware.a cpu/libcpu.a hardware/reSID/libresid.a fpu/libfpu.a gui/libgui.a output/liboutput.a hardware/mame/libmame.a libs/zmbv/libzmbv.a aviwriter/libaviwriter.a  xBRZ/libxbrz.a  -lasound -lm -ldl -lpthread -L/usr/local/lib -Wl,-rpath,/usr/local/lib -Wl,--enable-new-dtags -lSDL2 -ltinfo -lfreetype -lSDL_net -lz -lpng -lz -lpcap -lfluidsynth -lavcodec -lavformat -lavutil -lswscale -lavcodec -lGL
/usr/bin/ld: gui/libgui.a(midi.o): in function `MidiHandler_timidity::connect_to_server(char const*, int, _TCPsocket**)':
/home/pi/code/dosbox-x/src/gui/midi_timidity.h:219: undefined reference to `SDLNet_GetError'
/usr/bin/ld: /home/pi/code/dosbox-x/src/gui/midi_timidity.h:225: undefined reference to `SDLNet_GetError'
/usr/bin/ld: gui/libgui.a(midi.o): in function `MidiHandler_timidity::timidity_write_data(void const*, int)':
/home/pi/code/dosbox-x/src/gui/midi_timidity.h:326: undefined reference to `SDLNet_GetError'
/usr/bin/ld: gui/libgui.a(midi.o): in function `MidiHandler_timidity::timidity_ctl_command(char*, char const*, ...)':
/home/pi/code/dosbox-x/src/gui/midi_timidity.h:252: undefined reference to `SDLNet_GetError'
/usr/bin/ld: gui/libgui.a(midi.o): in function `MidiHandler_timidity::MidiHandler_timidity()':
/home/pi/code/dosbox-x/src/gui/midi_timidity.h:45: undefined reference to `SDLNet_GetError'
/usr/bin/ld: hardware/libhardware.a(ipx.o):/home/pi/code/dosbox-x/src/hardware/ipx.cpp:1104: more undefined references to `SDLNet_GetError' follow
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:471: dosbox-x] Error 1
make[3]: Leaving directory '/home/pi/code/dosbox-x/src'
make[2]: *** [Makefile:531: all-recursive] Error 1
make[2]: Leaving directory '/home/pi/code/dosbox-x/src'
make[1]: *** [Makefile:391: all-recursive] Error 1
make[1]: Leaving directory '/home/pi/code/dosbox-x'
make: *** [Makefile:328: all] Error 2

Thank-you, thank-you, thank-you! :)
Caroline

--- a/configure.ac
+++ b/configure.ac
@@ -392,7 +392,7 @@ LDFLAGS="$LDFLAGS -L$pwd/vs2015/sdlnet/linux-host/lib"
 CPPFLAGS="$CPPFLAGS -I$pwd/vs2015/sdlnet/linux-host/include -I$pwd/vs2015/sdlnet/linux-host/include/SDL"
 CXXFLAGS="$CXXFLAGS -I$pwd/vs2015/sdlnet/linux-host/include -I$pwd/vs2015/sdlnet/linux-host/include/SDL"
 
-dnl LIBRARY TEST: SDLnet 1.x initial test
+dnl LIBRARY TEST: SDLnet initial test
 AC_CHECK_HEADER(SDL_net.h,have_sdl_net_h=yes,)
 
 if test x$enable_emscripten != xyes; then
@@ -561,8 +561,9 @@ dnl This may be provided by glibc itself, -liconv may not be needed.
 AC_CHECK_HEADER(iconv.h,have_iconv_h=yes,)
 AC_CHECK_LIB(iconv, libiconv_open, have_iconv_lib=yes, ,-liconv)
 
-dnl LIBRARY TEST: SDLnet 1.x
+dnl LIBRARY TEST: SDLnet
 AC_CHECK_HEADER(SDL_net.h,have_sdl_net_h=yes,)
+AC_CHECK_LIB(SDL2_net, SDLNet_Init, have_sdl2_net_lib=yes, , )
 AC_CHECK_LIB(SDL_net, SDLNet_Init, have_sdl_net_lib=yes, , )
 
 AC_COMPILE_IFELSE([AC_LANG_SOURCE([
@@ -855,14 +856,30 @@ AH_TEMPLATE(C_SDL_NET,[Indicate whether SDL_net is present])
 AH_TEMPLATE(C_MODEM,[Define to 1 to enable internal modem support, requires SDL_net])
 AH_TEMPLATE(C_IPX,[Define to 1 to enable IPX over Internet networking, requires SDL_net])
 
-# HACK: Don't enable SDL2 net if SDL2
-if test x$have_sdl_net_lib = xyes -a x$have_sdl_net_h = xyes ; then
- LIBS="$LIBS -lSDL_net"
- AC_DEFINE(C_SDL_NET,1)
- AC_DEFINE(C_MODEM,1)
- AC_DEFINE(C_IPX,1)
-else 
- AC_MSG_WARN([Can't find SDL_net, internal modem and ipx disabled])
+dnl enable SDL2 net if SDL2
+if test -n "$SDL2_LIBS"; then
+ if test x$have_sdl2_net_lib = xyes -a x$have_sdl_net_h = xyes ; then
+  LIBS="$LIBS -lSDL2_net"
+  AC_DEFINE(C_SDL_NET,1)
+  AC_DEFINE(C_MODEM,1)
+  AC_DEFINE(C_IPX,1)
+ elif test x$have_sdl_net_lib = xyes -a x$have_sdl_net_h = xyes ; then
+  LIBS="$LIBS -lSDL_net"
+  AC_DEFINE(C_SDL_NET,1)
+  AC_DEFINE(C_MODEM,1)
+  AC_DEFINE(C_IPX,1)
+ else
+  AC_MSG_WARN([Can't find SDL_net, internal modem and ipx disabled])
+ fi
+else
+ if test x$have_sdl_net_lib = xyes -a x$have_sdl_net_h = xyes ; then
+  LIBS="$LIBS -lSDL_net"
+  AC_DEFINE(C_SDL_NET,1)
+  AC_DEFINE(C_MODEM,1)
+  AC_DEFINE(C_IPX,1)
+ else 
+  AC_MSG_WARN([Can't find SDL_net, internal modem and ipx disabled])
+ fi
 fi
 
 dnl FEATURE: Whether to support libz, and enable snapshots

